prow/cmd/deck/main.go

```diff
@@ -142,7 +144,79 @@ func staticHandlerFromDir(dir string) http.Handler {
 	return gziphandler.GzipHandler(handleCached(http.FileServer(http.Dir(dir))))
 }
 
+var (
+	prefixSet = sets.NewString("/tide.js",
+		"/tide-history.js",
+		"/plugin-help.js",
+		"/data.js",
+		"/prowjobs.js",
+		"/pr-data.js",
+		"/log",
+		"/rerun",
+		"/spyglass/static/",
+		"/spyglass/lens/",
+		"/view/",
+		"/job-history/",
+		"/pr-history/",
+	)
+	deckMetrics = struct {
+		httpRequestDuration *prometheus.HistogramVec
+	}{
+		httpRequestDuration: prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Name:    "deck_http_request_duration_seconds",
+				Help:    "http request duration seconds for deck server",
+				Buckets: []float64{0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1, 2.5, 5, 10},
```

Comment:
Do we want these to be configurable? I know in older versions of k8s the apiserver latency histogram tops out at 8s and this is frustrating since requests often take a lot longer.

Reply:
Can we do this with the next iteration?

Reply:
Yes